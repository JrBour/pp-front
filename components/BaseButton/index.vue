<template>
  <button
    :type="type"
    :disabled="disabled"
    :class="{ reduce: small }"
    :style="{ padding: loading ? '3px 0' : '1em 0' }"
    @click="handleClick"
  >
    <Loader v-if="loading" opacity="1" fill="#fff" height="50px" />
    <span v-else>{{ text }}</span>
  </button>
</template>

<script>
export default {
  props: {
    loading: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean
    },
    type: {
      type: String,
      default: 'button'
    },
    small: {
      type: Boolean,
      default: false
    },
    text: {
      type: String,
      required: true
    }
  },
  methods: {
    handleClick() {
      this.$emit('handle-click')
    }
  }
}
</script>

<style lang="scss" scoped>
button {
  box-sizing: content-box;
  width: 100%;
  border: none;
  border-radius: 10px;
  background: #3750b2;
  color: white;
  padding: 1em 0;
  font-weight: 500;
  font-size: 1.2em;
  margin: 4vh 0 2vh 0;

  &:disabled {
    opacity: 0.5;
  }
}
.reduce {
  display: block;
  width: 80%;
  margin: 3vh auto;
  font-size: 0.9em;
}
</style>
