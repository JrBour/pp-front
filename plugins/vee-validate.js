import { extend } from 'vee-validate'
import { required, email, max, min, numeric, alpha } from 'vee-validate/dist/rules'

extend('required', {
  ...required,
  message: 'Ce champ est requis'
})

extend('numeric', {
  ...numeric,
  message: 'Veuillez saisir des chiffres'
})

extend('alpha', {
  ...alpha,
  message: 'Veuillez ne saisir que des lettres'
})

extend('min', {
  ...min,
  validate(value, args) {
    return value.length >= args.length
  },
  params: ['length'],
  message: 'Ce champ doit contenir au moins {length} caracteres'
})

extend('max', {
  ...max,
  validate(value, args) {
    return value.length <= args.length
  },
  params: ['length'],
  message: 'Ce champ doit contenir au maximum {length} caracteres'
})

extend('email', {
  ...email,
  message: 'Email invalide'
})
