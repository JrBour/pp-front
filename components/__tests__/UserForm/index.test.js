import { shallowMount } from '@vue/test-utils'
import { ValidationObserver, ValidationProvider } from 'vee-validate'
import UserForm from '@/components/UserForm'
import BaseButton from '@/components/BaseButton'
import BaseInput from '@/components/BaseInput'
import Password from '@/components/Password'

describe('UserForm', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallowMount(UserForm, {
      stubs: {
        Password,
        BaseButton,
        BaseInput,
        ValidationObserver,
        ValidationProvider
      },
      propsData: {
        loading: false,
      },
    })
  })

  describe('computed', () => {
    it('should return false when all fields are filled correctly', async () => {
      await wrapper.setData({
        email: 'john@doe.com',
        firstname: 'John',
        lastname: 'Doe',
        password: 'P4ssword',
        phone: '0123456789',
      })

      expect(wrapper.vm.disabledSubmitButton).toBeFalsy()
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
