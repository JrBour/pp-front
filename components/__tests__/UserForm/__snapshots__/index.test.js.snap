// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm snapshot renders properly 1`] = `
<form><input id="profile" type="file" name="profile">
  <div class="register__profile" style="width: 65%;">
    <p>Photo <br>de profil</p> <label for="profile" class="register__profile_add">
      +
    </label>
  </div>
  <!---->
  <div class="input__wrapper" id="firstName">
    <div class="input__container"><input id="firstname" name="firstname" step="any" type="text" class=""> <label for="firstname" class="">
        Prenom <span>*</span></label></div>
    <p></p>
  </div>
  <div class="input__wrapper" id="name">
    <div class="input__container"><input id="lastname" name="lastname" step="any" type="text" class=""> <label for="lastname" class="">
        Nom <span>*</span></label></div>
    <p></p>
  </div>
  <div class="input__wrapper" id="phone">
    <div class="input__container"><input id="phone" name="phone" step="any" type="tel" class=""> <label for="phone" class="">
        Telephone <span>*</span></label></div>
    <p></p>
  </div>
  <div class="input__wrapper" id="email">
    <div class="input__container"><input id="email" name="email" step="any" type="email" class=""> <label for="email" class="">
        Email <span>*</span></label></div>
    <p></p>
  </div>
  <div class="input__wrapper">
    <div class="input__container"><input id="password" name="password" type="password" class=""> <label for="password" class="">Mot de passe <span>*</span></label> <img src="" alt="Eye icon">
      <!---->
    </div>
  </div>
  <!----> <button type="submit" disabled="disabled" class="" style="padding: 1em 0px;"><span>S'inscrire</span></button>
</form>
`;
